tools/pycafe/pycafe_utils.py

```diff
@@ -0,0 +1,158 @@
+"""Utility functions for generating PyCafe links."""
+
+import base64
+import gzip
+import json
+import textwrap
+from dataclasses import dataclass
+from typing import Optional
+from urllib.parse import quote, urlencode
+
+import requests
+import vizro
+from github import Auth, Github
+from github.Commit import Commit
+from github.Repository import Repository
+
+
+@dataclass
+class PyCafeConfig:
+    """Configuration for PyCafe link generation."""
+
+    github_token: str
+    repo_name: str
+    run_id: str
+    commit_sha: str
+    pr_number: Optional[int] = None
+    pycafe_url: str = "https://py.cafe"
+    vizro_raw_url: str = "https://raw.githubusercontent.com/mckinsey/vizro"
+    package_version: str = vizro.__version__
+
+
+def create_github_client(config: PyCafeConfig) -> tuple[Github, Repository, Commit]:
+    """Create GitHub client and return github, repo, and commit objects."""
+    auth = Auth.Token(config.github_token)
+    github = Github(auth=auth)
+    repo = github.get_repo(config.repo_name)
+    commit = repo.get_commit(config.commit_sha)
+    return github, repo, commit
```

Comment:
Why do we want the `github` part of this if it's not used when the function is called (I think)?